### PR TITLE
Import environment variables into envchain

### LINKED ISSUE #23: Implement envchain
### 💡 Feature/Idea

Right now environment variables (API keys etc.) are being stored in `home/dot_config/shell/private_private.sh.tmpl`. This is not exactly secure. We can improve security by implementing `envchain` (https://github.com/sorah/envchain).

Write a script in `home/.chezmoiscripts/universal/*.sh.tmpl` that scans the `private_private.sh.tmpl` file and imports all the variables into `envchain` and then deletes the `profile_profile.sh.tmpl` file. In environments where this is not possible with the system keyring, skip the `envchain` and leave the `profile_profile.sh.tmpl` file in  place.

Be sure to make the logic idempotent.

### :thumbsup: Can you contribute?

_No response_

### PR DESCRIPTION
**Pull request by enggnr**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This PR allows importing sensitive environment variables into `envchain`

- **What is the current behavior?** (You can also link to an open issue here)

- **What is the new behavior (if this is a feature change)?**

- **Other information**:
Fixes #23.

### CODE REVIEW

File: home/.chezmoiscripts/universal/run_before_01-import-secrets-envchain.tmpl [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{{- if (ne .host.distro.family "windows") -}}
+#!/usr/bin/env bash
+# @file Import environment variable values into `envchain`
+# @brief Ensures that values of sensitive environment variables are imported into `envchain` on either Linux or macOS
+# @description
+#     This script imports the values of sensitive environment variables into `envchain` if it is installed
```
Comment (ProfessorManhattan) [inline home/.chezmoiscripts/universal/run_before_01-import-secrets-envchain.tmpl:6]: Hey, can you add a bit more detail here on how it works?

Also, maybe we should cycle through the variables in the .chezmoitemplates/secrets folder instead to load them into envchain. That way the secrets are not stored anywhere on the disk in unencrypted format. We can get rid of the `private.sh` file (unless you think it's better this way then let me know what you think).

**Review by ProfessorManhattan — CHANGES_REQUESTED** [review]

File: home/.chezmoiscripts/universal/run_before_01-import-secrets-envchain.tmpl (hunk 1 above)
Comment (enggnr) [reply home/.chezmoiscripts/universal/run_before_01-import-secrets-envchain.tmpl:6]: Not having the keys in unencrypted form is better from a security standpoint. I saw in the documentation (or in the script templates) where we ask to source `private.sh`. We would need to update those instances to instead use `envchain <namespace> <executable>`.

A couple pain points:

* Each execution of the command needs to be prefixed with `envchain <namespace>`. Creating aliases may help with that. We can also provide instructions to set the variables from a namespace in the current shell
* The file also gives an easy way to load all values in one go. I do not think there is a way to load all the variables using `envchain`. This makes available what is absolutely needed to execute the program, which is good. Could we disable creation of `private.sh` unless opted for?

To add the secrets to envchain from the files, we need to name the files in such a way that we can detect the namespace from the name of the file. The part till the first underscore character can be taken as the name of the namespace and the entire file name can be the variable name. Existing files would work if we are to switch to the above method; some files need updating, e.g. `key-cloudflare-api-key`, `key-slack-app-client-id`, etc..

Comment (ProfessorManhattan) [reply home/.chezmoiscripts/universal/run_before_01-import-secrets-envchain.tmpl:6]: Hey @enggnr -- let's replace `private.sh` as an alias for `envchain` source import. Then, we can move all the variable declarations into a script file that lives right by the `private.sh` that imports all the variables into envchain. I think this is better than automatically importing all the variables from `.chezmoitemplates/secrets`.

Just import everything into the default collection.

Comment (ProfessorManhattan) [reply home/.chezmoiscripts/universal/run_before_01-import-secrets-envchain.tmpl:6]: Be sure to include documentation in both the `private.sh` file and the helper `after_script_script-name.sh.tmpl` file.

**Comment by ProfessorManhattan** [comment]
Hey @enggnr -- please get to this when you get a chance

**Comment by enggnr** [comment]
@ProfessorManhattan, helper file created. Please see the details in the my previous comment. Do we need 2 files for this feature?

**Comment by ProfessorManhattan** [comment]
Hey @enggnr -- sorry for the confusion on this one. Can you actually just create a new script in `.chezmoiscripts/universal` that goes through each variable in `.chezmoitemplates/secrets` and loads into the default context of the envchain keychain? That way, we can leave the `private.sh` file alone and eventually remove it.

**Review by enggnr — COMMENTED** [review]
Changes are in #63. This can be closed once that is merged.

**Pull request closed without merging**